ga4gh/datamodel/genotype_phenotype.py

```diff
@@ -17,47 +16,101 @@
 import ga4gh.datamodel as datamodel
 
 
-class AbstractG2PDataset(datamodel.DatamodelObject):
+class AbstractPhenotypeAssociationSet(datamodel.DatamodelObject):
+    compoundIdClass = datamodel.PhenotypeAssociationSetCompoundId
 
-    def __init__(self, setName=None):
-        self._searchQuery = ""
+    def __init__(self, parentContainer, localId):
+        super(AbstractPhenotypeAssociationSet, self).__init__(
+            parentContainer, localId)
 
     def toProtocolElement(self):
-        # TODO remove this... added to follow pattern of datadriven tests
+        pas = protocol.PhenotypeAssociationSet()
+        pas.name = self.getLocalId()
+        pas.id = self.getId()
+        pas.datasetId = self.getParentContainer().getId()
+        pas.info = self.getInfo()
+        return pas
+
+    def getInfo(self):
+        return {}
+
+    def _generateFeature(self, featureId="", childIds=[], parentId="",
+                         featureSetId="", referenceName="", start=None,
+                         end=None, strand=None, _term="", _termId="",
+                         sourceName="", sourceVersion=""):
+
+        feature = protocol.Feature()
+        feature.id = featureId
+        feature.childIds = []
+        feature.parentId = parentId
+        feature.featureSetId = featureSetId
+        feature.referenceName = referenceName
+        feature.start = start
+        feature.end = end
+        feature.strand = strand
+        feature.featureType = self._generateOntologyTerm(_term,
+                                                         _termId,
+                                                         sourceName,
+                                                         sourceVersion)
+        return feature
+
+    def _generatePhenotypeInstance(self, _id="", _term="",
+                                   _termId="", sourceName="",
+                                   sourceVersion="", qualifierTerms=None,
+                                   onsetTerm=None, description=""):
+        phenotypeInstance = protocol.PhenotypeInstance()
+        phenotypeInstance.id = _id
+        phenotypeInstance.type = self._generateOntologyTerm(_term,
+                                                            _termId,
+                                                            sourceName,
+                                                            sourceVersion)
+        phenotypeInstance.qualifier = qualifierTerms
+        phenotypeInstance.ageOfOnset = onsetTerm
+        phenotypeInstance.description = description
+        return phenotypeInstance
+
+    def _generateOntologyTerm(self, _term="", _id="",
```

Comment:
Why ``_term`` here? The ``_id`` convention is used here because ``id`` is a builtin function in Python and the convention used to avoid clashing with this identifier is to prefix it with an ``_``